cmd/skiff/top.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"archive/tar"
+	"container/heap"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/containers/image/v5/image"
+	"github.com/containers/image/v5/pkg/blobinfocache/none"
+	"github.com/containers/image/v5/pkg/compression"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	"github.com/urfave/cli/v3"
+)
+
+var topCommand = cli.Command{
+	Name:  "top",
+	Usage: "Analyze a container image and list files by size",
+	Flags: []cli.Flag{
+		&cli.BoolFlag{Name: "include-pseudo", Usage: "Include pseudo-filesystems (/dev, /proc, /sys)"},
+		&cli.BoolFlag{Name: "follow-symlinks", Usage: "Follow symbolic links"},
+	},
+	Arguments: []cli.Argument{
+		&cli.StringArg{Name: "image", Min: 1, Max: 1, Values: &url, UsageText: "Container image ref"},
+	},
+	Action: func(ctx context.Context, c *cli.Command) error {
+		sysCtx := types.SystemContext{}
+		return analyzeLayers(url[0], ctx, &sysCtx)
+	},
+}
+
+const defaultFileLimit = 10
+
+type FileInfo struct {
+	Path string
+	Size int64
+}
+
+type FileHeap []FileInfo
```

Comment:
This is not a heap, this is a stack

Reply:
What do you mean? are you referring to the data structure or the implementation on `FileHeap`? It's a min-heap implementation from https://pkg.go.dev/container/heap